presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1086,6 +1092,9 @@ protected Type visitLambdaExpression(LambdaExpression node, StackableAstVisitorC
             verify(types.size() == lambdaArguments.size());
 
             Map<String, LambdaArgumentDeclaration> nameToLambdaArgumentDeclarationMap = new HashMap<>();
+            if (context.getContext().isInLambda()) {
```

Comment:
Here and above. Perhaps this could be done unconditionally, if only `getNameToLambdaArgumentDeclarationMap()` returned an empty map when not already within lambda.